.github/workflows/esti.yaml

```diff
@@ -613,37 +637,88 @@ jobs:
         working-directory: test/spark
         run: ./setup-test.sh
 
-      - name: Copy repository ref
-        run: aws s3 cp --recursive s3://esti-system-testing-data/golden-files/gc-test-data s3://esti-system-testing/${{ github.run_number }}-spark${{ matrix.spark.version }}-metaclient/exporter
-
       - name: Setup GC tests
         env:
-          STORAGE_NAMESPACE: s3://esti-system-testing/${{ github.run_number }}-spark${{ matrix.spark.version }}-metaclient/garbage
+          STORAGE_NAMESPACE: s3://esti-system-testing/${{ github.run_number }}-spark${{ matrix.spark.version }}-metaclient/garbage/${{ steps.unique.outputs.value }}
           REPOSITORY: test-data-gc-${{ matrix.spark.version }}
         working-directory: test/spark
         run: ./setup-gc-test.sh
 
       - name: Test GC with Spark 3.x
         env:
-          STORAGE_NAMESPACE: s3://esti-system-testing/${{ github.run_number }}-spark${{ matrix.spark }}-metaclient/garbage
+          STORAGE_NAMESPACE: s3://esti-system-testing/${{ github.run_number }}-spark${{ matrix.spark }}-metaclient/garbage/${{ steps.unique.outputs.value }}
           REPOSITORY: test-data-gc-${{ matrix.spark.version }}
           CLIENT_JAR: ${{steps.import_metaclient.outputs.download-path}}/spark-assembly${{ matrix.spark.suffix }}.jar
         working-directory: test/spark
         run: ./run-gc-test.sh
 
+      - name: lakeFS Logs on Spark with gateway failure
+        if: ${{ failure() }}
+        continue-on-error: true
+        working-directory: test/spark
+        run: docker-compose logs --tail=1000 lakefs
+
+  metadata-client-export-spark3:
```

Comment:
New test? Is this related to this PR?

Reply:
Sort of, but mostly in order to pass tests.  In eb8df2b19048ef8173b6e1874b79db30b1e0331b I split the metadata client test into gc and export so that they may run in parallel -- that lets the export test (just a few minutes) run without timing out, and also helps the long-running gc test finish running on time.
Without this, Esti barely managed to sneak in under the deadline -- which would have meant we cannot trust this PR.
I shall update the PR description to reflect this addition. :-)  So now we won't have a dirty hack, we will have a **documented** dirty hack.